packages/svelte/src/internal/client/types.d.ts

```diff
@@ -108,7 +117,7 @@ export type ComputationSignal<V = unknown> = {
 	/** references: Anything that a signal owns */
 	r: null | ComputationSignal[];
 	/** value: The latest value for this signal, doubles as the teardown for effects */
-	v: V;
+	v: V | DerivedSignalState<V>;
```

Comment:
This feels quite confusing to me — the fact that it forces us to cast in various places feels like a bit of a code smell. Can we just add another property?

Reply:
Two thoughts here went through my mind when developing this:

- If we add another field, then all computational signal objects pay a memory size cost. Profiling it, the cost was actually way bigger than I was happy with.
- Using an existing field and overloading it alleviated the memory cost from the previous point at the cost of more types and runtime complexity.

Reply:
by 'all computational signals' do you mean effects and deriveds? in that case i have to wonder if they shouldn't just be separate shapes — it feels like we'd have fewer compromises that way